libs/failsafe-controller/src/main/java/net/jodah/failsafe/FailsafeController.java

```diff
@@ -405,34 +498,27 @@ private synchronized void onCompletion(R result, Throwable error) {
     executions.currentExecution().ifPresent(exec -> exec.onCompletion(result, error));
   }
 
+  static <T> ContextualCallable<T> callableOf(final ContextualRunnable runnable) {
+    Assert.notNull(runnable, FailsafeController.RUNNABLE);
+    return c -> {
+      runnable.run(c);
+      return null;
+    };
+  }
+
   /**
    * This class is used to intercept all failsafe configuration in order to allow us to control
    * them. Any sync or async created later will always delegate configuration to this class as the
    * master which will allow us to keep one copy of the configuration and interceptors.
    */
-  class SyncBuilder extends SyncFailsafe<R> {
+  class SyncBuilder extends SyncFailsafeConfigDelegater<R> {
```

Comment:
ℹ️ Extending the delegate allows the sync to inherit the ability to register asynchronous listeners that can be monitored when invoked.